x-pack/packages/kbn-elastic-assistant/impl/assistant/use_send_message/index.tsx

```diff
@@ -9,29 +9,29 @@ import { HttpSetup } from '@kbn/core-http-browser';
 import { useCallback, useState } from 'react';
 
 import { useAssistantContext } from '../../assistant_context';
-import { Conversation, Message } from '../../assistant_context/types';
+import { Conversation } from '../../assistant_context/types';
 import { fetchConnectorExecuteAction, FetchConnectorExecuteResponse } from '../api';
 
-interface SendMessagesProps {
+interface SendMessageProps {
   allow?: string[];
   allowReplacement?: string[];
   apiConfig: Conversation['apiConfig'];
   http: HttpSetup;
-  messages: Message[];
-  onNewReplacements: (newReplacements: Record<string, string>) => void;
-  replacements?: Record<string, string>;
+  message?: string;
```

Comment:
should it be possible for `message` to be undefined?

Reply:
yes, this would happen when we hit "Regenerate message". Regenerate message is accomplished by deleting the last message of the conversation, and triggering the `sendMessage` API without the `message`. Then in `post_actions_connector_execute` we will send the previous messages only to the llm:
```
messages: [...(prevMessages ?? []), ...(newMessage ? [newMessage] : [])],
```

We should handle the case where both prevMessages and newMessage are undefined. This should not happen with our UI, but we should handle with an error message anyways. Add it to the to do list